lib/dotcom_web/components/trip_planner/itinerary_group.ex

```diff
@@ -4,18 +4,24 @@ defmodule DotcomWeb.Components.TripPlanner.ItineraryGroup do
   """
   use DotcomWeb, :component
 
-  import DotcomWeb.Components.TripPlanner.ItineraryDetail
-
   attr(:summary, :map, doc: "ItineraryGroups.summary()", required: true)
   attr(:itineraries, :list, doc: "List of %Dotcom.TripPlan.Itinerary{}", required: true)
 
+  attr(:index, :integer,
```

Comment:
It took me a while to understand what this was for... ultimately I'd favor a more descriptive attribute name, maybe `:group_index`?

Reply:
Thinking for a moment... the only reason an itinerary group cares about its `index` is because it does _the right thing_ in response to clicking on `Details`. So maybe the thing to do is have setting `index` correctly be handled in a callback instead of implicitly through this param.

I think that would require converting `itinerary_group` to a live component though, so if we think that's a good idea, my vote would be to do that as a follow-up PR.

Reply:
I think that would work fine (and agree that might need a live component), but it might be overkill? 

An alternate thought: the summary and itineraries here are populated by the output of `Dotcom.TripPlan.ItineraryGroups.from_itineraries/1`, which outputs a list of `%{itineraries: itineraries, summary: summary}` maps. Why not have this function compute the indexes and have the map include that `index` as well?

A more meta thought - we're using index to really mean "unique identifier". The past iteration made up a unique ID (`assign(assigns, :group_id, "group-#{:erlang.phash2(assigns.itineraries)}")`). We might want to reconsider using the index value for that purpose (what if we want to let end users re-sort the list later, for example).